src/frequenz/client/microgrid/sensor.py

```diff
@@ -48,3 +73,170 @@ def __str__(self) -> str:
         """Return a human-readable string representation of this instance."""
         name = f":{self.name}" if self.name else ""
         return f"<{type(self).__name__}:{self.id}{name}>"
+
+
+@enum.unique
+class SensorMetric(enum.Enum):
+    """The metrics that can be reported by sensors in the microgrid.
+
+    These metrics correspond to various sensor readings primarily related to
+    environmental conditions and physical measurements.
+    """
+
+    UNSPECIFIED = 0
+    """Default value (this should not be normally used and usually indicates an issue)."""
+
+    TEMPERATURE = 1
+    """Temperature, in Celsius (°C)."""
+
+    HUMIDITY = 2
+    """Humidity, in percentage (%)."""
+
+    PRESSURE = 3
+    """Pressure, in Pascal (Pa)."""
+
+    IRRADIANCE = 4
+    """Irradiance / Radiation flux, in watts per square meter (W / m²)."""
+
+    VELOCITY = 5
+    """Velocity, in meters per second (m / s)."""
+
+    ACCELERATION = 6
+    """Acceleration in meters per second per second (m / s²)."""
+
+    ANGLE = 7
+    """Angle, in degrees with respect to the (magnetic) North (°)."""
+
+    DEW_POINT = 8
+    """Dew point, in Celsius (°C).
+
+    The temperature at which the air becomes saturated with water vapor.
+    """
+
+
+@enum.unique
+class SensorStateCode(enum.Enum):
+    """The various states that a sensor can be in."""
+
+    UNSPECIFIED = 0
+    """Default value (this should not be normally used and usually indicates an issue)."""
+
+    ON = 1
+    """The sensor is up and running."""
+
+    ERROR = 2
+    """The sensor is in an error state."""
+
+
+@enum.unique
+class SensorErrorCode(enum.Enum):
+    """The various errors that can occur in sensors."""
+
+    UNSPECIFIED = 0
+    """Default value (this should not be normally used and usually indicates an issue)."""
+
+    UNKNOWN = 1
+    """An unknown or undefined error.
+
+    This is used when the error can be retrieved from the sensor but it doesn't match
+    any known error or can't be interpreted for some reason.
+    """
+
+    INTERNAL = 2
+    """An internal error within the sensor."""
+
+
+@dataclass(frozen=True, kw_only=True)
+class SensorStateSample:
+    """A sample of state, warnings, and errors for a sensor at a specific time."""
+
+    sampled_at: datetime
```

Comment:
wasn't it supposed to be `sample_time`?

Reply:
No, actually the current candidate is `snapshot_time` (and the message will be `SensorStateSnapshot`), but it is not changed yet, for stuff that it is already a bit unstable/changing, I just left the name used in 0.17. @tiyash-basu-frequenz do you think this is stable enough to adopt the new names in the Python API already?

Reply:
Thinking about this, there are **many** things that will change, even if they are considered fairly stable now, I think I prefer to leave the "modernization" of the client interface for a follow-up PR, let's not block this one on that as we are supposed to have people wanting to use this now.

Reply:
* https://github.com/frequenz-floss/frequenz-client-microgrid-python/issues/147